code/game/objects/items/devices/scanners/plant_analyzer.dm

```diff
@@ -0,0 +1,275 @@
+		graft = target
+	else if (isitem(target))
+		var/obj/item/scanned_object = target
+		if(istype(scanned_object, /obj/item/seeds))
+			seed = scanned_object
+		else if (scanned_object.get_plant_seed())
+			seed = scanned_object.get_plant_seed()
+
+	if (!seed && !tray && !graft)
+		return FALSE
+
+	last_scan_data = list(
+		"tray_data" = null,
+		"seed_data" = null,
+		"graft_data" = null,
+	)
+
+	if(tray)
+		last_scan_data["tray_data"] = list(
+			"plant_health" = tray.plant_health,
+			"plant_age" = tray.age,
+			"name" = tray.name,
+			"icon" = tray.icon,
+			"icon_state" = tray.icon_state,
+			"water" = tray.waterlevel,
+			"water_max" = tray.maxwater,
+			"nutri" = tray.reagents.total_volume,
+			"nutri_max" = tray.maxnutri,
+			"yield_mod" = tray.yieldmod,
+			"weeds" = tray.weedlevel,
+			"weeds_max" = MAX_TRAY_WEEDS,
+			"pests" = tray.pestlevel,
+			"pests_max" = MAX_TRAY_PESTS,
+			"toxins" = tray.toxic,
+			"toxins_max" = MAX_TRAY_TOXINS,
+			"reagents" = list(),
+		)
+		for(var/datum/reagent/reagent in tray.reagents.reagent_list)
+			last_scan_data["tray_data"]["reagents"] += list(list(
+				"name" = reagent.name,
+				"volume" = reagent.volume
+			))
+
+	if(seed)
+		last_scan_data["seed_data"] = make_seed_data(seed)
+
+	if(graft)
+		last_scan_data["graft_data"] = list(
+			"name" = graft.parent_name,
+			"icon" = graft.icon,
+			"icon_state" = graft.icon_state,
+			"yield" = graft.yield,
+			"production" = graft.production,
+			"lifespan" = graft.lifespan,
+			"endurance" = graft.endurance,
+			"weed_rate" = graft.weed_rate,
+			"weed_chance" = graft.weed_chance,
+			"graft_gene" = graft.stored_trait.type
+		)
+
+	ui_interact(user)
+
+	return TRUE
+
+
+/obj/item/plant_analyzer/proc/make_seed_data(obj/item/seeds/seed)
+	var/list/seed_data = list(
+		"name" = seed.plantname,
+		"icon" = seed.growing_icon,
+		"icon_state" = seed.icon_harvest,
+		"product" = seed.product?.name,
+		"product_icon" = seed.product?.icon,
+		"product_icon_state" = seed.product?.icon_state,
+		"potency" = seed.potency,
+		"yield" = seed.yield,
+		"instability" = seed.instability,
+		"maturation" = seed.maturation,
+		"production" = seed.production,
+		"lifespan" = seed.lifespan,
+		"endurance" = seed.endurance,
+		"weed_rate" = seed.weed_rate,
+		"weed_chance" = seed.weed_chance,
+		"graft_gene" = seed.graft_gene ? seed.graft_gene : /datum/plant_gene/trait/repeated_harvest
+	)
+	seed_data["removable_traits"] = list()
+	seed_data["core_traits"] = list()
+	for(var/datum/plant_gene/trait/trait in seed.genes)
+		if(trait.mutability_flags & PLANT_GENE_REMOVABLE)
+			seed_data["removable_traits"] += trait.type
+		else
+			seed_data["core_traits"] += trait.type
+	seed_data["reagents"] = list()
+	for(var/datum/plant_gene/reagent/reagent in seed.genes)
+		seed_data["reagents"] += list(list(
+			"name" = reagent.name,
+			"rate" = reagent.rate
+		))
+	seed_data["volume_mod"] = (locate(/datum/plant_gene/trait/maxchem) in seed.genes) ? 2 : 1
```

Comment:
The `2` isn't a static number, it's based on the trait's `rate` var, so it should grab it from there